driver/ixsystems/common.py

```diff
@@ -339,7 +339,14 @@ def _delete_volume(self, name):
             fullvolume, snapname = clone.split('@')
             temp, snapvol = fullvolume.rsplit('/', 1)
             self._delete_snapshot(snapname, snapvol)
-        if ret['status'] != FreeNASServer.STATUS_OK:
+
+        # When deleting volume with dependent snapsnot clone, 422 error triggered. Throw VolumeIsBus exception ensures
+        # upper stream cinder manager mark volume status available instead of error-deleting.
+        if ret['status'] == 'error' and ret['response'] == '422:Unprocessable Entity':
+            errorexception = exception.VolumeIsBusy(
+                _("Cannot delete volume when clone child volume or snapshot exists!"), volume_name=name)
+            raise errorexception
+        elif ret['status'] != FreeNASServer.STATUS_OK:
             msg = ('Error while deleting volume: %s' % ret['response'])
```

Comment:
For https://github.com/iXsystems/cinder/issues/24, special exception VolumeIsBusy raise to upper stream cinder manager in case of dependency clone caused by create volume from volume, create volume from snapshot. In this case raise VolumeIsBusy allows upper stream cinder driver return safely, the delete volume does not stuck into error-deleting status.
After fix, when user trying to delete a volume that has cloned volume, the action simply return, volume still stay in available status, without stuck into error-delete. I tried to pop up and display a more user friendly error message but limit to current upstream code, no luck so far. This maybe enhanced later on but might heavily depends on upstream code.